### PR TITLE
Test userAgent for Safari string

### PR DESCRIPTION
**Pull request by webdeb**

### CODE REVIEW

**Comment by eligrey** [comment]
No user agent sniffing.

**Comment by webdeb** [comment]
@eligrey in our case the `view.safari` prop was undefined (iPad mini 10.1), and this check helped, however, could you explain why this is not a good idea?

**Comment by eligrey** [comment]
Oh, if the previous check wasn't working please mention so.

**Comment by webdeb** [comment]
thanks, I'll take a look what we can do to be more precisely.

**Comment by webdeb** [comment]
But actually, it should be fine, `CriOS` is tested the same way and `Safari` is not like `Mozilla` which is almost everywhere ;)

**Comment by eligrey** [comment]
This gives a false positive on the most popular browser on the internet, Google Chrome.

**Comment by webdeb** [comment]
oh..

**Comment by webdeb** [comment]
updated, I've used this https://stackoverflow.com/a/23522755/5004923 answer in combo with the `CriOS` check.

**Comment by Ruffio** [comment]
@eligrey can this PR be merged or are there issues with it?

File: FileSaver.js [hunk 1]
```diff
@@ -31,8 +31,11 @@ var saveAs = saveAs || (function(view) {
 			var event = new MouseEvent("click");
 			node.dispatchEvent(event);
 		}
-		, is_safari = /constructor/i.test(view.HTMLElement) || view.safari
 		, is_chrome_ios =/CriOS\/[\d]+/.test(navigator.userAgent)
+		, is_safari = !is_chrome_ios &&
```
Comment (ogonkov) [inline FileSaver.js]: Chrome on iOS is just wrapper around Safari core, isn't it?

**Comment by Ruffio** [comment]
@webdeb is this still relevant?

**Comment by Ruffio** [comment]
@eligrey this PR should be rejected as it is incomplete and the author has failed to respond to ping for three months...

**Comment by Ruffio** [comment]
@eligrey shouldn't this PR be closed due to inactivity/incomplete ?

File: FileSaver.js (hunk 1 above)
Comment (Ruffio) [reply FileSaver.js]: I don't think that you are correct about that. They both use webkit as engine, from there to say that is it just a wrapper I think is a long way

**Comment by Ruffio** [comment]
What is it that we are trying to fix with this PR? I mean is FileSaver broken in Safari browsers?

I can see that you use the 'negative look-around' which I think it the correct way to do it in order to exclude false positives from Chrome browsers. From Mozilla's site we have:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent

image

@eligrey what do you think?

**Comment by Ruffio** [comment]
I also found this list of user-agents that is pretty complete: https://deviceatlas.com/blog/mobile-browser-user-agent-strings

Btw. maybe it would be a good idea to implement some tests for 'correct' userAgent testing in the project? The above list is a good starting point.

**Pull request closed without merging**